clients/go/coreutils/literals.go

```diff
@@ -483,7 +484,17 @@ func MakeLiteralForType(t *core.LiteralType, v interface{}) (*core.Literal, erro
 		}
 	case *core.LiteralType_Simple:
 		newT := t.Type.(*core.LiteralType_Simple)
-		lv, err := MakeLiteralForSimpleType(newT.Simple, fmt.Sprintf("%v", v))
+		strValue := fmt.Sprintf("%v", v)
+		if newT.Simple == core.SimpleType_STRUCT {
+			if _, isValueStringType := v.(string); !isValueStringType {
```

Comment:
Just for my own learning - if it is of SimpleType_STRUCT, why could it be a string?

Reply:
MakeLiteralForType(t *core.LiteralType, v interface{})  accepts the type and value as an interface and hence its possible for the client of this function to pass in the value as json string directly instead of this function taking care of it. If the string is not in a proper format , it will complain during deserialization.

I have added a unit test `Generic Passed As String`  to explain this .